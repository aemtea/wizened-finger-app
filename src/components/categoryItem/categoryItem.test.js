/**
 * @jest-environment jsdom
 */

 import React from 'react';
 import { render, fireEvent } from '@testing-library/react-native';
 import renderer from 'react-test-renderer';
 import CategoryItem from '../../components/categoryItem/categoryItem';
 
 describe('<CategoryItem />', () => {
   let mockFn;
   const category = {
     id: 0,
     title: 'hello world'
   };
 
   beforeEach(() => {
     mockFn = jest.fn();
   });
 
   it('onPress returns item', () => {
     const { getByText } = render(<CategoryItem item={category} onCategorySelected={mockFn} />);
 
     fireEvent.press(getByText(category.title));
     
     expect(mockFn).toBeCalledWith(category);
   });
 
   it('renders correctly', () => {
     const tree = renderer.create(<CategoryItem item={category} onCategorySelected={mockFn} />).toJSON();
     expect(tree).toMatchSnapshot();
   });
 });