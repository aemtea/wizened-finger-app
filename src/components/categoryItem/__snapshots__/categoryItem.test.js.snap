// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryItem /> renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Text
    style={
      Object {
        "fontSize": 32,
      }
    }
  >
    hello world
  </Text>
</View>
`;
